### PR TITLE
#3151 Power with an option for negative input values

### PR DESCRIPTION
**Pull request by ErwanGuichard49**
closes [#3151](https://github.com/dynawo/dynawo/issues/3151)

### CODE REVIEW

**Review by joyelfeghali — APPROVED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/Power.mo [hunk 1]
```diff
@@ -9,20 +9,29 @@ within Dynawo.NonElectrical.Blocks.Continuous;
...
-* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite
+* of simulation tools for power systems.
 */
 
 block Power "Outputs a power of the input"
   extends Modelica.Blocks.Interfaces.SISO;
 
   parameter Real N "Exponent of the power function";
+  parameter Boolean NInteger = true "For u < 0 : if true, u ^ N is calculable, if false, u ^ N is not calculable, y = 0 instead";
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/Power.mo:20]: Having a default value = true, makes it possible to change it to false later on?

**Review by joyelfeghali — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/Power.mo (hunk 1 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/Power.mo:20]: In ExTkjEdf1.mo (dynawo-rte branch 753), NInteger is false, the simulations are successful.

A default value can be overriden if the block is used in a Modelica model, this is different from changing the value from a .par file.

Comment (joyelfeghali) [reply dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/Power.mo:20]: But if we change it in the .par, we are not sure if we get the right value

Comment (rosiereflo) [reply dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/Power.mo:20]: The simulation can be successful but the model behaves as if this parameter was true... better avoid default values.

Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/Power.mo]: The default value has been removed.

**Review by joyelfeghali — APPROVED** [review]

**Pull request merged**